rfcs/2022014-layernorm-v2/README.md

```diff
@@ -0,0 +1,175 @@
+for int8 inference.
+
+## Proposal
+
+### Option 1
+
+There are no options to cover different memory descriptors and algorithms other
+than creating a new operation descriptor, `dnnl_layer_normalization_v2_desc_t`.
+The prior descriptor `dnnl_layer_normalization_desc_t` will also need to be
+renamed `dnnl_layer_normalization_v1_desc_t`, but would otherwise be unchanged.
+The new descriptor will be extended over previous one with `dnnl_memory_desc_t
+dst_desc` and `dnnl_memory_desc_t diff_dst_desc`. `data_desc` analogues will be
+replaced with `src_desc` correspondently. Implementation details will be similar
+to pooling_v2 and softmax_v2. We will also mark `layer_normalization_v1` API as
+deprecated, but in docs and release notes only. No code changes to not frustrate
+our users dealing with warning and potential errors (with DNNL_WERROR=ON).
+
+#### C API
+
+```c
+/// dnnl_types.h
+
+/// Kinds of primitives. Used to implement a way to extend the library with new
+/// primitives without changing the ABI.
+typedef enum {
+    ...
+    /// A layer normalization version 2 primitive (layer normalization with
+    /// destination memory descriptor and algorithm kind).
+    dnnl_layer_normalization_v2,
+    ...
+} dnnl_primitive_kind_t;
+
+/// A descriptor of a Layer Normalization operation.
+typedef struct {
+    /// The kind of primitive. Used for self-identifying the primitive
+    /// descriptor. Must be #dnnl_layer_normalization.
+    dnnl_primitive_kind_t primitive_kind;
+    /// The kind of propagation. Possible values: #dnnl_forward_training,
+    /// #dnnl_forward_inference, #dnnl_backward, and #dnnl_backward_data.
+    dnnl_prop_kind_t prop_kind;
+    /// Source memory descriptor.
+    dnnl_memory_desc_t src_desc;
+    /// Source gradient memory descriptor.
+    dnnl_memory_desc_t diff_src_desc;
+    /// Scale and shift data and gradient memory descriptors.
+    ///
+    /// Scaleshift memory descriptor uses 2D #dnnl_ab
+    /// format[2, normalized_dim] where 1-st dimension contains gamma parameter,
+    /// 2-nd dimension contains beta parameter. Normalized_dim is equal to the
+    /// last logical dimension of the data tensor across which normalization is
+    /// performed.
+    dnnl_memory_desc_t data_scaleshift_desc;
+    dnnl_memory_desc_t diff_data_scaleshift_desc;
+    /// Mean and variance data memory descriptors.
+    ///
+    /// Statistics (mean and variance) memory descriptor is the k-dimensional
+    /// tensor where k is equal to data_tensor_ndims - 1 and may have any plain
+    /// (stride[last_dim] == 1) user-provided format.
+    dnnl_memory_desc_t stat_desc;
+    /// Destination memory descriptor.
+    dnnl_memory_desc_t dst_desc;
+    /// Destination gradient memory descriptor.
+    dnnl_memory_desc_t diff_dst_desc;
+    /// Layer normalization epsilon parameter.
+    float layer_norm_epsilon;
+    unsigned flags;
+} dnnl_layer_normalization_desc_t;
```

Comment:
I believe this structure should be renamed to not break ABI for the existing one
```suggestion
} dnnl_layer_normalization_desc_v2_t;
```

Reply:
Good catch, thanks! It's been changed to match the introductory option 1 text.